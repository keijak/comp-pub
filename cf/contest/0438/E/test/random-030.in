235 101
778
790
373
348
683
902
95
536
687
788
365
235
276
214
787
130
199
418
23
985
42
123
973
995
481
948
527
158
270
292
452
45
3
16
71
690
377
180
176
878
474
21
962
997
382
835
2
700
139
751
663
52
850
956
150
225
967
40
473
725
755
666
78
297
695
637
25
932
579
384
415
861
24
173
674
378
201
392
342
490
555
709
465
409
456
346
183
886
500
545
607
41
698
318
434
232
164
1
352
922
255
485
820
914
406
526
447
65
592
656
640
275
525
93
841
691
574
264
806
589
541
309
905
132
964
404
408
335
60
881
776
593
194
144
1000
928
450
626
55
332
800
207
421
509
814
215
679
446
442
107
133
617
87
844
848
245
893
127
475
62
295
251
491
47
221
998
819
773
472
662
258
355
390
334
926
875
36
957
570
519
888
621
678
412
198
839
552
111
11
991
858
478
879
461
853
623
208
594
388
191
585
569
547
911
627
772
745
780
854
939
752
14
887
971
227
913
518
821
705
537
568
624
369
868
869
117
145
7
946
386
597
172
375
767
90
