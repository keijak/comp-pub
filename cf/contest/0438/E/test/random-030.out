1
3
10
35
132
519
2106
8750
37052
159319
693767
3053292
13559572
60687984
273464085
241352577
657454229
905198123
108937865
697826294
652834287
913740127
335498772
924281860
764485739
809852631
510088653
503744634
649158124
527915264
561961648
770231628
486313970
925363367
942435906
894515476
349561624
323366850
116208143
962991745
574767919
598334513
726180017
239529425
760489312
306406576
944286015
472014372
976152808
955308520
85693300
795483868
102410587
903332068
474618308
102950162
165548003
753351267
51005571
430372584
613611722
250212741
596458669
269455326
770952745
921619445
618508055
972685673
231391943
961755464
115306602
806370571
709914754
247568094
242221130
670693113
712249109
206433425
307450105
706030874
372906909
113818975
753032206
320308099
193767746
153711177
422504768
445172925
422028767
944563896
340658485
589081038
438970443
953726332
379995020
11803712
327532669
488188936
982999123
1360973
958734688
